### PR TITLE
fix: javascript error on PWA preview

### PR DESCRIPTION
**Pull request by N-Feist**
[[AB#106152](https://dev.azure.com/plentymarkets/0af57df9-8662-4901-bb97-8e88b07ff0c9/_workitems/edit/106152)]

## Why:

Closes: #ID

## Checklist before requesting a review

- [ ] My code follows the code style of this project.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I've updated `docs/changelog/changelog_en.md` and `docs/changelog/changelog_de.md`. If I'm a non-German speaker, I've still updated the file with the English version as a placeholder.

### CODE REVIEW

File: apps/web/sdk/index.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ export const useSdk = () => {
   const sdkConfig = {
     // @ts-ignore
     plentysystems: buildModule<PlentysystemsModuleType>(plentysystemsModule, {
-      apiUrl: `${config.public.apiUrl}/plentysystems`,
+      apiUrl: `${config.public.apiUrl}/plentysystems`.replace(/[/\\]+$/, ''),
```
Comment (maxiroellplenty) [inline apps/web/sdk/index.ts]: i dont think we need the check here.

File: apps/web/nuxt.config.ts [hunk 2]
```diff
@@ -174,7 +174,7 @@ export default defineNuxtConfig({
   },
   runtimeConfig: {
     public: {
-      apiUrl: process.env.API_URL ?? 'http://localhost:8181',
+      apiUrl: (process.env.API_URL ?? 'http://localhost:8181').replace(/[/\\]+$/, ''),
```
Comment (maxiroellplenty) [inline apps/web/nuxt.config.ts]: we could write a helper here that gets us the api url and that can be tested.

Comment (maxiroellplenty) [reply apps/web/nuxt.config.ts]: test what happens if we have a / at the end
test what happens if we have two slashes at the end
test what happens if we dont have a / at the end

**Review by maxiroellplenty — CHANGES_REQUESTED** [review]

File: apps/web/nuxt.config.ts (hunk 2 above)
Comment (N-Feist) [reply apps/web/nuxt.config.ts]: @maxiroellplenty just pushed the changes

File: apps/web/sdk/index.ts (hunk 1 above)
Comment (N-Feist) [reply apps/web/sdk/index.ts]: @maxiroellplenty removed it in last commit

File: apps/web/nuxt.config.ts [hunk 3]
```diff
@@ -174,7 +175,7 @@ export default defineNuxtConfig({
   },
   runtimeConfig: {
     public: {
-      apiUrl: process.env.API_URL ?? 'http://localhost:8181',
+      apiUrl: removeTrailingSlash(process.env.API_URL),
```
Comment (maxiroellplenty) [inline apps/web/nuxt.config.ts]: would rename that function to something like `validateApiUrl` oder `getApiUrl` and if we make it validateApiUrl I would extract the localhost:8181 address back to were it was before.

**Review by maxiroellplenty — CHANGES_REQUESTED** [review]

File: docs/changelog/changelog_de.md [hunk 4]
```diff
@@ -13,6 +13,7 @@
 
 ### Behoben
 
+- javascript Error in der PWA Vorschau
```
Comment (maxiroellplenty) [inline docs/changelog/changelog_de.md]:
```suggestion
- Es wird sicher gestellt, dass sich kein `/` am Ende der API_URL befindet.
```

File: docs/changelog/changelog_en.md [hunk 5]
```diff
@@ -14,6 +14,7 @@
 
 ### Fixed
 
+- javascript error on PWA preview
```
Comment (maxiroellplenty) [inline docs/changelog/changelog_en.md]:
```suggestion
- We do now ensure that there is no `/` at the end of the API_URL.
```

**Review by maxiroellplenty — CHANGES_REQUESTED** [review]

File: apps/web/__tests__/test/unit/apiUrl.spec.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import { describe, expect, it } from 'vitest'
+import { validateApiUrl } from '../../../utils/pathHelper';
+
+describe('removeTrailingSlash function', () => {
```
Comment (maxiroellplenty) [inline apps/web/__tests__/test/unit/apiUrl.spec.ts]:
```suggestion
describe('validateApiUrl function', () => {
```

**Review by maxiroellplenty — CHANGES_REQUESTED** [review]

**Review by maxiroellplenty — APPROVED** [review]

**Pull request merged**